tests/codebuild/create_tests.sh

```diff
@@ -80,15 +80,21 @@ function verify_debug_test
 {
   local crd_type="$1"
   local crd_instance="$2"
-  local expected_debug_job_status="$3"
+  local timeout="$3"
+  local expected_debug_job_status="$4"
+  # First verify that trainingjob has been completed
+  verify_test TrainingJob xgboost-mnist-debugger $timeout Completed
+
   # TODO extend this for multiple debug job with debug job statuses parameter
-  local found_debug_job_status="$(kubectl get "$crd_type" "$crd_instance" -o json | jq -r '.status.debugRuleEvaluationStatuses[0].ruleEvaluationStatus')"
-  if [ "${expected_debug_job_status}" == "${found_debug_job_status}" ]; then
-      echo "[PASSED] Verified ${crd_type} ${crd_instance} debug job has status ${expected_debug_job_status}"
-  else
-      echo "[FAILED] ${crd_type} ${crd_instance} debug job has status ${found_debug_job_status} while expected status was ${expected_debug_job_status}"
-      exit 1
-  fi
+
+  echo "Waiting for debug job to finish"
+  timeout "${timeout}" bash -c \
+    'until [ "$(kubectl get "$0" "$1" -o json | jq -r .status.debugRuleEvaluationStatuses[0].ruleEvaluationStatus)" != "$3" ]; do \
+      echo "Debug job has not completed yet"; \
+      sleep 1; \
+    done' "${crd_type}" "${crd_instance}"
```

Comment:
You need a catch block here, otherwise even if it times out nothing will happen.

```
if [ $? -ne 0 ]; then
  echo "[FAILED] ..."
  exit 1
fi
```